sips/core/msg_struct_encoding.md

```diff
@@ -35,6 +35,52 @@ SSZ is 66% smaller (in the test, 1,640 bytes vs 3,700) vs json, using snappy com
 
 SSZ hash tree root enables substituting full data objects with their hash tree root without changing the full message hash tree root which also doesn’t change the BLS signature signing the object. Below we explain how this property is used.
 
+**Compression**  
+We’ve chosen to use [S2](https://github.com/klauspost/compress/tree/master/s2#snappy-compatibility) compression for SSZ encoding, based on the benchmark described below.
+
+S2 offers full compatibility with Snappy.
+This means that the efficient encoders of S2 can be used to generate fully Snappy compatible output.
+
+***size comparison between different encoding/compression methods:***
+```
+ssz:            1220
+ssz snappy:     156
+ssz s2:         134
+ssz s2 snappy:  161
+json:           2860
+json snappy:    476
+json s2:        407
+json s2 snappy: 487
+```
+
+***Benchmark***
+```
+go test -bench=. ./ssz_encoding/qbft -benchmem -benchtime 15s -cpu 2
+
+goos: darwin
+goarch: amd64
+pkg: ssv-experiments/ssz_encoding/qbft
+cpu: Intel(R) Core(TM) i9-9880H CPU @ 2.30GHz
+--------------------------------------------------------------------------------------------------------------------------
+NAME-Number of cores                            Iterations           Nanoseconds/op        Number of bytes     Number of alloccs
```

Comment:
@olegshmuelov can you also benchmark plain JSON here? as we need to compare to current baseline (JSON)

Reply:
the benchmark comparison for the JSON is in the table

Reply:
I mean w/o compression (like we have today)